### PR TITLE
Hw spring

### PR DESCRIPTION
**Pull request by frostpv**

### CODE REVIEW

File: src/main/java/models/User.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+@Entity
+@Table (name = "users")
+public class User {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String name;
+    private String email;
+    private int age;
+}
```
Comment (vladkuzya) [inline src/main/java/models/User.java:22]: What about setters/getters?

**Review by vladkuzya — COMMENTED** [review]
Nice work!)

File: src/main/java/models/User.java (hunk 1 above)
Comment (frostpv) [reply src/main/java/models/User.java:22]: I used Lombok lib it generated automatically.

File: src/main/java/dao/UserDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class UserDaoImpl implements UserDao {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
```
Comment (boroda4436) [inline src/main/java/dao/UserDaoImpl.java]: Before Spring 4.3, we had to add an @Autowired annotation to the constructor. With newer versions, this is optional if the class has only one constructor.

Comment (frostpv) [reply src/main/java/dao/UserDaoImpl.java]: it looks like magic. )))

File: src/main/java/dao/UserDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(user);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert user "
+                    + user + " into data", e);
```
Comment (RomanDubovskyi) [inline src/main/java/dao/UserDaoImpl.java]:
```suggestion
                    + user + " into db", e);
```
data might be a bit to abstract

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**